### PR TITLE
Update Upgrade-Guide with respect to R4

### PR DESCRIPTION
**Pull request by matofeder**
R4 upgrade-guide

### CODE REVIEW

File: doc/Upgrade-Guide.md [hunk 1]
```diff
@@ -150,17 +157,54 @@ to tweak both `clusterctl.yaml` and `cluster-template.yaml` for the
...
+
+In the R3 to R4 process, also `cloud.conf` added `enable-ingress-hostname=true` to the
+LoadBalancer section. It is due to `NGINX_INGRESS_PROXY` defaulting to true now. So if
+you want to use, or you are already using this proxy functionality we recommend you to
+add this line to your `cloud.conf`, e.g.:
+```bash
+echo "enable-ingress-hostname=true" >> ~/cluster-defaults/cloud.conf
+echo "enable-ingress-hostname=true" >> ~/testcluster/cloud.conf
+```
+Then, before upgrading workload cluster by `create.sh`,
```
Comment (garloff) [inline doc/Upgrade-Guide.md:186]: by `create-cluster.sh`

**Review by garloff — APPROVED** [review]
Your currently proposed doc is a clear improvement, I just saw one typo (create-*cluster*.sh).
As you marked it WIP, I don't want to just merge it, as you may have some things in mind you want to complete today or tomorrow before merging.
I suggest not waiting for too long to avoid conflicts.

File: doc/Upgrade-Guide.md (hunk 1 above)
Comment (matofeder) [reply doc/Upgrade-Guide.md]: addressed by a3f2b20c8e43d7be27437045a9bbd75a922c97d4, thx @garloff !

**Comment by matofeder** [comment]
> As you marked it WIP, I don't want to just merge it, as you may have some things in mind you want to complete today or tomorrow before merging. I suggest not waiting for too long to avoid conflicts.

@garloff thanks for the review & typo fix & rebase!

WIP/Draft status has been removed. This PR can be merged.

**Pull request merged**